skbio/alignment/_tabular_msa.py

```diff
@@ -1701,7 +1701,15 @@ def reassign_index(self, mapping=None, minter=None):
                 "Cannot use both `mapping` and `minter` at the same time.")
 
         if mapping is not None:
-            self._seqs.rename(mapping, inplace=True)
+            if isinstance(mapping, dict):
+                self.index = [mapping[label] if label in mapping else label
+                              for label in self.index]
+            elif callable(mapping):
+                self.index = [mapping(label) for label in self.index]
+            else:
```

Comment:
This turned out very simple...

Now isn't there an edge case with respect to `tuples` and `pd.MultiIndex`? I don't remember if assignment always converted to a MultiIndex where possible. I'll double check that.